### DISCUSSION TITLE
Members in batch list is blank nothing is coming in dropdown

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#557] [Sunbird ED configuration discussion] [answered]
Hello All,

I am trying to enable the invite-only batch feature on Sunbird-Ed. 

I have added a button for switching the batches and it opens the members in the batch list dropdown.

The dropdown got enabled by itself when I added the button code.

Here is the button code

```
<div class="sb-radio-btn-checkbox sb-radio-btn-primary py-8">
         <input id="invite-only" type="radio" value='invite-only' name="enrollmentType" formControlName="enrollmentType">
         <label for="invite-only">{{resourceService?.frmelmnts?.lbl?.inviteOnly}}</label>
</div>
```

But no values are coming in the participant dropdown,
I checked the code but did not understand much about how it is getting the participant list.
I am clearly missing some parts. 

Please guide me here.

![membersbatchempty](https://github.com/Sunbird-Ed/Community/assets/65657373/7f121854-219a-4a56-9300-ae9b83be0fca)

cc: @coolbung @Renuka-S

**Comment by Xitija** [comment]
Can anyone please help me with this? I am unable to proceed further

**Comment by chitranshu-keshav** [comment]
@rajeevsathish @vinukumar-vs can you please get back on this

**Comment by rajeevsathish** [comment]
@Xitija in the code we have blocked the invite-only batch and the participantList. so please enable the same so that you can see the value in the participantList list dropdown

cc: @chitranshu-keshav @vinukumar-vs

**Reply by Xitija** [reply]
@rajeevsathish Sir,
Can you please point me to where this change has to be made?
It might be obvious, but I am fairly new to this.

I am currently on Tag release-4.10.0_RC1, is this feature enabled in this release, if yes then please let me know how should I enable it .
If not which branch should I refer for the invite-only feature, and where can we see the code in that branch (as you mentioned that code is blocked)

**Reply by Xitija** [reply]
@rajeevsathish Sir,
I checked further the getUserList() method in CourseBatchService is calling the API for getting only the mentor list.
What can be done further to fetch all the users?

cc: @chitranshu-keshav  @coolbung

**Reply by Xitija** [reply]
I need to understand this on an urgent basis. It is very critical. 
Please help me.

cc: @chitranshu-keshav @rajeevsathish  @coolbung

**Reply by princegupta1131** [reply]
Hi @Xitija we are using the below API call to get the user list based on the roles.

```
https://staging.sunbirded.org/learner/user/v3/search
{
    "request": {
        "filters": {
            "status": "1",
            "rootOrgId": "01269878797503692810",
            "organisations.roles": [
                "COURSE_MENTOR"
            ]
        },
        "query": ""
    }
}
```
**Can you try this request? instead of "COURSE_MENTOR" pass empty and try**
```
{
    "request": {
        "filters": {
            "status": "1",
            "rootOrgId": "{{rootorgId}}",
            "organisations.roles": [
               
            ]
        },
        "query": ""
    }
}
```
@vinukumar-vs  can you please confirm on this 

CC: @chitranshu-keshav @rajeevsathish  @vinukumar-vs

**Comment by Xitija** [comment]
@princegupta1131

I changed the API body by removing 
`option.data.request.filters['organisations.roles'] = ['COURSE_MENTOR'];`
as you have suggested.

It is getting all the users in the current organization, but there needs to be some code that will set the data to the dropdown on the change of batches.
In the current scenario when we click on the create batch, a form is opened the form's value is set to 'open' so it does not enable the dropdown for the participants (only mentors can be selected), and when the batch type is changed through the form to 'invite-only' the participant dropdown gets enabled but there is no event that will set the value to the participant list in the UI even though data from API is fetched.

If you have more information on this please guide me.

**Comment by Krishnaj20** [comment]
@rajeevsathish - Can you look into this thread and take it to closure by providing the correct advise? 

@vinukumar-vs @chitranshu-keshav

**Reply by Krishnaj20** [reply]
@rajeevsathish - Is this thread closed? If not, please propose a sync call to explain the code changes needed to be done by @Xitija . Please target closure asap.

**Reply by Xitija** [reply, marked as answer]
Hello All,

I referred the code from Sunbird 2.9 branch which previously had invite-only feature implemented.
But I could not find the solution to the problem stated above.
With support from the team, I figured out that it was happening because an external jquery script `semanticDropdown.js` is added.
We managed to make it work with some code changes.
